python/pyiceberg/catalog/sql.py

```diff
@@ -0,0 +1,448 @@
+        """
+        database_name, table_name = self.identifier_to_database_and_table(identifier, NoSuchTableError)
+        with Session(self.engine) as session:
+            res = session.execute(
+                delete(IcebergTables).where(
+                    IcebergTables.catalog_name == self.name,
+                    IcebergTables.table_namespace == database_name,
+                    IcebergTables.table_name == table_name,
+                )
+            )
+            session.commit()
+        if res.rowcount < 1:
+            raise NoSuchTableError(f"Table does not exist: {database_name}.{table_name}")
+
+    def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: Union[str, Identifier]) -> Table:
+        """Rename a fully classified table name.
+
+        Args:
+            from_identifier (str | Identifier): Existing table identifier.
+            to_identifier (str | Identifier): New table identifier.
+
+        Returns:
+            Table: the updated table instance with its metadata.
+
+        Raises:
+            NoSuchTableError: If a table with the name does not exist.
+            TableAlreadyExistsError: If a table with the new name already exist.
+            NoSuchNamespaceError: If the target namespace does not exist.
+        """
+        from_database_name, from_table_name = self.identifier_to_database_and_table(from_identifier, NoSuchTableError)
+        to_database_name, to_table_name = self.identifier_to_database_and_table(to_identifier)
+        if not self._namespace_exists(to_database_name):
+            raise NoSuchNamespaceError(f"Namespace does not exist: {to_database_name}")
+        with Session(self.engine) as session:
+            try:
+                stmt = (
+                    update(IcebergTables)
+                    .where(
+                        IcebergTables.catalog_name == self.name,
+                        IcebergTables.table_namespace == from_database_name,
+                        IcebergTables.table_name == from_table_name,
+                    )
+                    .values(table_namespace=to_database_name, table_name=to_table_name)
+                )
+                result = session.execute(stmt)
+                if result.rowcount < 1:
+                    raise NoSuchTableError(f"Table does not exist: {from_table_name}")
+                session.commit()
+            except IntegrityError as e:
+                raise TableAlreadyExistsError(f"Table {to_database_name}.{to_table_name} already exists") from e
+        return self.load_table(to_identifier)
+
+    def _namespace_exists(self, identifier: Union[str, Identifier]) -> bool:
+        namespace = self.identifier_to_database(identifier)
+        with Session(self.engine) as session:
+            stmt = (
+                select(IcebergTables)
+                .where(IcebergTables.catalog_name == self.name, IcebergTables.table_namespace.like(namespace + "%"))
```

Comment:
I don't think we want to have the `startsWith` here. 
```suggestion
                .where(IcebergTables.catalog_name == self.name, IcebergTables.table_namespace == namespace)
```

For example, this is used for the `create_namespace`. If you would create a namespace `production`, then you can't create a database `prod` anymore. Furthermore, the `%` doesn't play well with indexes.